### PR TITLE
Upload build time-trace data to CI database

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
The compiler's profile data (`-ftime-trace`) is uploaded to ClickHouse Cloud.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit d977e221b42c74dc131e8e741bff86eb1ab3f146 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53100/d977e221b42c74dc131e8e741bff86eb1ab3f146/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docker/test/style/Dockerfile [hunk 1]
```diff
@@ -19,7 +19,8 @@ RUN apt-get update && env DEBIAN_FRONTEND=noninteractive apt-get install --yes \
     shellcheck \
     yamllint \
     locales \
-    && pip3 install black==23.1.0 boto3 codespell==2.2.1 mypy==1.3.0 PyGithub unidiff pylint==2.6.2 \
+    && pip3 install black==23.1.0 boto3 codespell==2.2.1 mypy==1.3.0 PyGithub unidiff pylint==2.6.2 types-requests=2.31.0.2 types-urllib3=1.26.25.14 \
```
Comment (Felixoid) [inline docker/test/style/Dockerfile:22]: it's unnecessary change, we don't use additional stubs

File: tests/ci/build_check.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+from typing import List, Tuple
 import subprocess
 import logging
 import json
 import os
 import sys
 import time
-from typing import List, Tuple
+import urllib.parse
+import requests
```
Comment (Felixoid) [inline tests/ci/build_check.py:11]: I think, the goal is to use the same DB as for checks, so https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/clickhouse_helper.py#L18 would be extended

**Review by Felixoid — CHANGES_REQUESTED** [review]

File: .github/workflows/backport_branches.yml [hunk 3]
```diff
@@ -3,6 +3,9 @@ name: BackportPR
 env:
   # Force the stdout and stderr streams to be unbuffered
   PYTHONUNBUFFERED: 1
+  # Export system tables to ClickHouse Cloud
+  CLICKHOUSE_CI_LOGS_HOST: ${{ secrets.CLICKHOUSE_CI_LOGS_HOST }}
```
Comment (Felixoid) [inline .github/workflows/backport_branches.yml:7]: We usually use the SSM for this purpose, and this function https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/get_robot_token.py#L18

File: tests/ci/build_check.py (hunk 2 above)
Comment (alexey-milovidov) [reply tests/ci/build_check.py:11]: No, we will use a separate DB because the DB with checks is capped at about 3.6 TB, while ClickHouse Cloud is unlimited.

File: docker/test/style/Dockerfile (hunk 1 above)
Comment (alexey-milovidov) [reply docker/test/style/Dockerfile:22]: But `mypy` told me about it in clear text.

**Comment by alexey-milovidov** [comment]
Ok. It works, and the amount of changes is minimal.
Next step: change it to use SSM agent, but without breaking it.

**Comment by Felixoid** [comment]
I don't get the purpose of the review if it's just ignored completely and merged.

A new image has broken everything again
